### ISSUE TITLE
Exif data not parsed for specific images

### ISSUE THREAD

**Issue by bh213** [mattiasw/ExifReader#58]
### Description

EXIF data is not parsed for specific JPEG files. I am trying to extract color profile info and used the files from here: http://regex.info/blog/photo-tech/color-spaces-page2

For example this image: http://regex.info/i/cs/Momiji-sRGB-yes.jpg

```
exiftool.exe .\Momiji-sRGB-yes.jpg

ExifTool Version Number         : 11.65
File Name                       : Momiji-sRGB-yes.jpg
Directory                       : .
File Size                       : 46 kB
File Modification Date/Time     : 2019:09:30 09:55:48+02:00
File Access Date/Time           : 2019:09:30 09:55:48+02:00
File Creation Date/Time         : 2019:09:30 09:55:47+02:00
File Permissions                : rw-rw-rw-
File Type                       : JPEG
File Type Extension             : jpg
MIME Type                       : image/jpeg
JFIF Version                    : 1.02
Resolution Unit                 : None
X Resolution                    : 100
Y Resolution                    : 100
Quality                         : 30%
Profile CMM Type                : Linotronic
Profile Version                 : 2.1.0
Profile Class                   : Display Device Profile
Color Space Data                : RGB
Profile Connection Space        : XYZ
Profile Date Time               : 1998:02:09 06:49:00
Profile File Signature          : acsp
Primary Platform                : Microsoft Corporation
CMM Flags                       : Not Embedded, Independent
Device Manufacturer             : Hewlett-Packard
Device Model                    : sRGB
Device Attributes               : Reflective, Glossy, Positive, Color
Rendering Intent                : Perceptual
Connection Space Illuminant     : 0.9642 1 0.82491
Profile Creator                 : Hewlett-Packard
Profile ID                      : 0
Profile Copyright               : Copyright (c) 1998 Hewlett-Packard Company
Profile Description             : sRGB IEC61966-2.1
Media White Point               : 0.95045 1 1.08905
Media Black Point               : 0 0 0
Red Matrix Column               : 0.43607 0.22249 0.01392
Green Matrix Column             : 0.38515 0.71687 0.09708
Blue Matrix Column              : 0.14307 0.06061 0.7141
Device Mfg Desc                 : IEC http://www.iec.ch
Device Model Desc               : IEC 61966-2.1 Default RGB colour space - sRGB
Viewing Cond Desc               : Reference Viewing Condition in IEC61966-2.1
Viewing Cond Illuminant         : 19.6445 20.3718 16.8089
Viewing Cond Surround           : 3.92889 4.07439 3.36179
Viewing Cond Illuminant Type    : D50
Luminance                       : 76.03647 80 87.12462
Measurement Observer            : CIE 1931
Measurement Backing             : 0 0 0
Measurement Geometry            : Unknown
Measurement Flare               : 0.999%
Measurement Illuminant          : D65
Technology                      : Cathode Ray Tube Display
Red Tone Reproduction Curve     : (Binary data 2060 bytes, use -b option to extract)
Green Tone Reproduction Curve   : (Binary data 2060 bytes, use -b option to extract)
Blue Tone Reproduction Curve    : (Binary data 2060 bytes, use -b option to extract)
DCT Encode Version              : 100
APP14 Flags 0                   : [14], Encoded with Blend=1 downsampling
APP14 Flags 1                   : (none)
Color Transform                 : YCbCr
Image Width                     : 600
Image Height                    : 450
Encoding Process                : Progressive DCT, Huffman coding
Bits Per Sample                 : 8
Color Components                : 3
Y Cb Cr Sub Sampling            : YCbCr4:2:0 (2 2)
Image Size                      : 600x450
Megapixels                      : 0.270

```

What I get from ExifReader either in my code or by using examples/global/index.html:
```
Bits Per Sample	8
Image Height	450px
Image Width	600px
Color Components	3
Subsampling	YCbCr4:2:0 (2 2)
```

Quickly checking source code, I see that `fileDataOffset` is set but other fields ( `tiffHeaderOffset`, 
 `iptcDataOffset`, ...) are not.

**Comment by bh213** [comment]
Seems this jpeg uses JFIF, not sure if this is something this library is supposed to support?

**Comment by mattiasw** [comment]
Hi! No problem with JFIF, however, from what I understand most of those fields are from the ICC color profile which is not supported. I haven't looked at it enough to know how much work it would be to implement it, but I guess it would be possible. Will have a deeper look later on.

**Comment by bh213** [comment]
@mattiasw I mostly added ICC profile parsing to this library, let me know if you would be interesting in accepting PR with these changes.

Current work is here (in master, I'll squash commits later): https://github.com/bh213/ExifReader/commits/master

I still have to refactor tags parsing into separate class with better names, add a few tests & update typescript definitions.

**Comment by mattiasw** [comment]
Nice! I would definitely be interested in a PR for those changes.

**Comment by bh213** [comment]
I created a PR, please take a look, at very least I'll have to add  a few tests...

**Comment by mattiasw** [comment]
I'll try to have a look over the weekend.

**Comment by mattiasw** [comment]
Implemented in #59.
